### DISCUSSION TITLE
How to create multiple variant for Text

### DISCUSSION THREAD

**Discussion by hansyulian** [mantinedev/mantine#5853] [Q&A] [answered]
Hi, so far i saw <Text> and <Title> for the typography, i would like to know you what is the most recommended way to handle multiple variant/type of text such as `subtitle`, `dateTime`, `label` while using the mantine theme override.
what come in my mind is some kind of props like `<Text variant='label'>`, but i would like to know your suggestions

**Comment by rtivital** [comment, marked as answer]
https://mantine.dev/styles/variants-sizes/

**Reply by hansyulian** [reply]
hey, i implement the variant for text, i'm at lost on how to make the typescript able to know the options for variants, if you would help point me to the doc/example would be very appreciated. thanks!

**Reply by rtivital** [reply]
As far as I know, it is not possible.

**Reply by tombohub** [reply]
Wrap text in your own component and have variant props

**Comment by andoHolto** [comment]
**By no means is this stress-tested** (obviously comes with its own risks), but the following has allowed me to add to the variant union type for a Button using the [declaration merging / augmentation guide for colors](https://mantine.dev/theming/colors/#add-custom-colors-types) as a basis:


```typescript
import type { ButtonFactory, StylesApiProps } from '@mantine/core'

declare module '@mantine/core' {
  export interface ButtonProps {
    variant?: StylesApiProps<ButtonFactory>['variant'] | 'my-variant'
  }
}
```
<img width="318" alt="image" src="https://github.com/user-attachments/assets/12665005-ec37-48cc-87b1-cbcd0c645b2f">

**Comment by aigre87** [comment]
something like that,
// local mantine.d.ts
```
type ExtendedTextVariant = MantineTextTextVariant | 'newTextVariant';

declare module '@mantine/core' {
  export interface TextProps {
    variant?: ExtendedTextVariant;
  }
}
```

but in general, unfortunately, it looks like everything depends on the component that you are extending, if the prop that you want to extend is hardcoded in the mantine core types, then it seems that you can only type it by wrapping mantine component in your own component...
for example i cant define types for: 

 -Notification variant: // core Notification.d.ts
```
 export interface NotificationProps extends BoxProps, StylesApiProps<NotificationFactory>, ElementProps<'div', 'title'> {
    variant?: string;
    ...
```
-or redefine Button sizes because - // core Button.d.ts 
```
export interface ButtonProps extends BoxProps, StylesApiProps<ButtonFactory> {
    /** Controls button `height`, `font-size` and horizontal `padding`, `'sm'` by default */
    size?: MantineSize | `compact-${MantineSize}` | (string & {});
    ...
```
